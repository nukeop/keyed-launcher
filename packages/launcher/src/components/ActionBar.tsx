import { FC, ReactNode } from 'react';
import { PerformanceDashboard } from './PerformanceDashboard';

export type ActionBarProps = {
  icon?: string | ReactNode;
};

export const ActionBar: FC<ActionBarProps> = ({ icon }) => {
  return (
    <div
      className="flex w-full items-center justify-between border-t border-white/20 bg-black/20 p-2 text-white"
      data-testid="action-bar"
    >
      <div>{icon}</div>
      <PerformanceDashboard />
    </div>
  );
};
