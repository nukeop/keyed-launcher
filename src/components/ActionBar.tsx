import { FC, ReactNode } from 'react';

export type ActionBarProps = {
  icon?: string | ReactNode;
};

export const ActionBar: FC<ActionBarProps> = ({ icon }) => {
  return (
    <div className="flex w-full items-center justify-between border-t border-white/20 bg-black/20 p-2 text-white">
      <div>{icon}</div>
      <div className="text-xs">Test</div>
    </div>
  );
};
