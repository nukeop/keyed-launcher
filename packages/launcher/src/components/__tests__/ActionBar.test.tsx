import { ActionBar } from '../ActionBar';
import { render, screen, waitFor } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

vi.mock('../../utils/environment', () => ({
  isDev: vi.fn(() => false),
  isProd: vi.fn(() => false), // Set to false so performance dashboard renders
}));

vi.mock('../../utils/performance', () => ({
  PerformanceMonitor: {
    getCurrentFPS: vi.fn(() => 60),
    getMemoryStats: vi.fn(() =>
      Promise.resolve({
        used: '128 MB',
        total: '8 GB',
      }),
    ),
  },
}));

describe('ActionBar', () => {
  it('renders with default props', () => {
    render(<ActionBar />);

    expect(screen.getByTestId('performance-dashboard')).toBeInTheDocument();
  });

  it('renders with icon prop as string', () => {
    render(<ActionBar icon="⚙️" />);

    expect(screen.getByText('⚙️')).toBeInTheDocument();
    expect(screen.getByTestId('performance-dashboard')).toBeInTheDocument();
  });

  it('renders with icon prop as ReactNode', () => {
    const IconComponent = () => (
      <span data-testid="custom-icon">Custom Icon</span>
    );
    render(<ActionBar icon={<IconComponent />} />);

    expect(screen.getByTestId('custom-icon')).toBeInTheDocument();
    expect(screen.getByText('Custom Icon')).toBeInTheDocument();
    expect(screen.getByTestId('performance-dashboard')).toBeInTheDocument();
  });

  it('matches snapshot', async () => {
    const { container } = render(<ActionBar icon="⚙️" />);
    await waitFor(() => {
      expect(screen.getByTestId('performance-dashboard')).toBeInTheDocument();
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot without icon', async () => {
    const { container } = render(<ActionBar />);
    await waitFor(() => {
      expect(screen.getByTestId('performance-dashboard')).toBeInTheDocument();
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders performance dashboard with stats', async () => {
    render(<ActionBar icon="⚙️" />);

    const performanceDashboard = await screen.findByTestId(
      'performance-dashboard',
    );

    expect(performanceDashboard).toHaveTextContent('FPS60RAM128 MB/ 8 GB');
  });
});
