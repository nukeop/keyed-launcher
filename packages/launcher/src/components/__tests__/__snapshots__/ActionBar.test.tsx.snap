// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionBar > matches snapshot 1`] = `
<div
  class="flex w-full items-center justify-between border-t border-white/20 bg-black/20 p-2 text-white"
  data-testid="action-bar"
>
  <div>
    ⚙️
  </div>
  <div
    class="select-none"
    data-testid="performance-dashboard"
  >
    <div
      class="font-mono text-xs text-white"
    >
      <div
        class="flex items-center gap-4"
      >
        <div
          class="flex items-center gap-2"
          data-testid="fps"
        >
          <span
            class="font-bold text-yellow-400"
          >
            FPS
          </span>
          <span
            class="font-bold text-green-400"
          >
            60
          </span>
        </div>
        <div
          class="flex items-center gap-2"
          data-testid="ram"
        >
          <span
            class="font-bold text-blue-400"
          >
            RAM
          </span>
          <span
            class="text-white"
          >
            128 MB
          </span>
          <span
            class="text-gray-400"
          >
            / 
            8 GB
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ActionBar > matches snapshot without icon 1`] = `
<div
  class="flex w-full items-center justify-between border-t border-white/20 bg-black/20 p-2 text-white"
  data-testid="action-bar"
>
  <div />
  <div
    class="select-none"
    data-testid="performance-dashboard"
  >
    <div
      class="font-mono text-xs text-white"
    >
      <div
        class="flex items-center gap-4"
      >
        <div
          class="flex items-center gap-2"
          data-testid="fps"
        >
          <span
            class="font-bold text-yellow-400"
          >
            FPS
          </span>
          <span
            class="font-bold text-green-400"
          >
            60
          </span>
        </div>
        <div
          class="flex items-center gap-2"
          data-testid="ram"
        >
          <span
            class="font-bold text-blue-400"
          >
            RAM
          </span>
          <span
            class="text-white"
          >
            128 MB
          </span>
          <span
            class="text-gray-400"
          >
            / 
            8 GB
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
